packages/components/src/tools-panel/test/index.js

```diff
@@ -983,4 +983,56 @@ describe( 'ToolsPanel', () => {
 			expect( optionsDisplayedIcon ).toBeInTheDocument();
 		} );
 	} );
+
+	describe( 'first and last panel items', () => {
+		it( 'should apply first/last classes to appropriate items', () => {
+			render(
+				<SlotFillProvider>
+					<ToolsPanelItems>
+						<ToolsPanelItem { ...altControlProps }>
+							<div>Item 1</div>
+						</ToolsPanelItem>
+						<ToolsPanelItem { ...controlProps }>
+							<div>Item 2</div>
+						</ToolsPanelItem>
+					</ToolsPanelItems>
+					<ToolsPanelItems>
+						<ToolsPanelItem
+							{ ...altControlProps }
+							label="Item 3"
+							isShownByDefault={ true }
+						>
+							<div>Item 3</div>
+						</ToolsPanelItem>
+					</ToolsPanelItems>
+					<ToolsPanelItems>
+						<ToolsPanelItem { ...altControlProps } label="Item 4">
+							<div>Item 4</div>
+						</ToolsPanelItem>
+					</ToolsPanelItems>
+					<ToolsPanel
+						{ ...defaultProps }
+						hasInnerWrapper={ true }
+						shouldRenderPlaceholderItems={ true }
+					>
+						<Slot />
+					</ToolsPanel>
+				</SlotFillProvider>
+			);
+
+			const item2 = screen.getByText( 'Item 2' );
+			const item3 = screen.getByText( 'Item 3' );
+
+			expect( screen.queryByText( 'Item 1' ) ).not.toBeInTheDocument();
+			expect( item2 ).toBeInTheDocument();
+			expect( item3 ).toBeInTheDocument();
+			expect( screen.queryByText( 'Item 4' ) ).not.toBeInTheDocument();
+
+			expect( item2.parentNode ).toHaveClass( 'first' );
+			expect( item2.parentNode ).not.toHaveClass( 'last' );
+
+			expect( item3.parentNode ).not.toHaveClass( 'first' );
+			expect( item3.parentNode ).toHaveClass( 'last' );
```

Comment:
As @aaronrobertshaw mentioned earlier in this PR, we should avoid testing implementation details like classnames (actually, we should not use hardcoded classnames at all).

Instead, our tests should reflect the way a user interacts with the component. In this specific case, though, I can't really think of a way to do that, especially because no styles are really being applied.

I'd say that probably the best approach here is to use `toMatchSnapshot()` or `toMatchDiffSnapshot()` (as explained in the [testing overview](https://github.com/WordPress/gutenberg/blob/trunk/docs/contributors/code/testing-overview.md#best-practices))

Reply:
This test has been updated to use `toMatchSnapshot` in 5b3ae82

Reply:
Further updates will be made once the approach to first/last class names is settled.